crates/biome_service/src/workspace/server.rs

```diff
@@ -75,7 +75,7 @@ pub(crate) struct Document {
     pub(crate) file_source_index: usize,
 
     /// The result of the parser (syntax tree + diagnostics).
-    pub(crate) syntax: AnyParse,
+    pub(crate) syntax: Result<AnyParse, FileTooLarge>,
```

Comment:
I think at this point we should start thinking about splitting `WorkspaceError` into two types:
- `WorkspaceError`: actual errors that can occur during the call of a workspace API (missing method, serialisation mistakes, failing to call the server, etc.
- `WorkspaceDiagnostics`: a new type, where we collect all diagnostics during the lifecycle of an API (parsing errors, file too large, file ignored, etc.). These diagnostics aren't errors and shouldn't abruptly stop an API call, contrary to `WorkspaceError`

What do you think?

Reply:
Yeah agreed, that’s a good suggestion 👍 Only reason I didn’t yet is because we only have a single kind for now, but if we need to have multiple diagnostics I’ll do it like this.